src/components/CreateCourse/CreateCourse.tsx

```diff
@@ -0,0 +1,195 @@
+import { useMemo, useState } from 'react';
+import { Button } from '../../common/Button/Button';
+import { Input } from '../../common/Input/Input';
+import { AuthorItem } from './components/AuthorItem/AuthorItem';
+import { IAuthor, ICourse } from '../../mockData';
+import { getCourseDuration } from '../../helpers';
+import styles from './CreateCourse.module.css';
+import { v4 as uuidv4 } from 'uuid';
+
+interface ICreateCourseProps {
+	allAppAuthors: IAuthor[];
+	handleAddAuthor: (name: string) => void;
+	handleAddNewCourse: (course: ICourse) => void;
+}
+
+export const CreateCourse = ({
+	allAppAuthors,
+	handleAddAuthor,
+	handleAddNewCourse,
+}: ICreateCourseProps) => {
+	const [title, setTitle] = useState('');
+	const [description, setDescription] = useState('');
+	const [descriptionError, setDescriptionError] = useState(false);
+	const [courseAuthors, setCourseAuthors] = useState<IAuthor[]>([]);
+	const [createAuthor, setCreateAuthor] = useState('');
+	const [duration, setDuration] = useState<number | ''>('');
```

Comment:
avoid mixing primitives when possible, using `number | undefined` would be more appropriate here